### DISCUSSION TITLE
Intercept validation error messages in the controller

### DISCUSSION THREAD

**Discussion by OgaBoss** [spatie/laravel-data#446] [Q&A] [answered]
I have a very simple controller to handle login. it looks like this 

![Screenshot 2023-05-15 at 22 17 35](https://github.com/spatie/laravel-data/assets/16806901/52289035-e237-4942-b422-59693d805603)

Below is the DTO class
![Screenshot 2023-05-15 at 22 18 24](https://github.com/spatie/laravel-data/assets/16806901/6d69134a-ea9e-4297-9f9c-3d0ba4194ee9)

When i run the controller and i dont pass the required data i get response like this
`
{
    "message": "The email field is required. (and 1 more error)",
    "errors": {
        "email": [
            "The email field is required."
        ],
        "password": [
            "The password field is required."
        ]
    }
}
`

Now my question is this, is there a way to intercept this error, so i can reformat, before sending to the frontend ?
Thank you for creating this wonderful package.

**Comment by rubenvanassche** [comment, marked as answer]
We're using the Laravel validator which throws a ValidationException, catching that one and reformatting that one and retrowing it again should probably do the trick.
